src/test/java/com/epam/test_generator/controllers/TestResultControllerTest.java

```diff
@@ -0,0 +1,252 @@
+package com.epam.test_generator.controllers;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.epam.test_generator.dto.RawSuitResultDTO;
+import com.epam.test_generator.dto.TestResultDTO;
+import com.epam.test_generator.entities.Status;
+import com.epam.test_generator.services.TestResultService;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.http.MediaType;
+import org.springframework.security.core.Authentication;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestResultControllerTest {
+
+    private static final long SIMPLE_PROJECT_ID = 0L;
+    private static final Integer FROM = 2;
+    private static final Integer TO = 3;
+    private static final Integer NEGATIVE = -5;
+    private static final DateTimeFormatter dtf = DateTimeFormatter.ISO_DATE;
+    private static final SimpleDateFormat format = new SimpleDateFormat("dd/mm/yyyy");
+    private static Date FROM_DATE;
+    private static Date TO_DATE;
+
+    private ObjectMapper mapper = new ObjectMapper();
+    private MockMvc mockMvc;
+    private TestResultDTO testResultDTO;
+    private List<TestResultDTO> testResultDTOS;
+    private List<RawSuitResultDTO> rawSuitResultDTOS;
+
+    @Mock
+    private TestResultService testResultService;
+
+    @InjectMocks
+    private TestResultController testResultController;
+
+    @Before
+    public void setup() throws ParseException {
+        mockMvc = MockMvcBuilders.standaloneSetup(testResultController)
+            .setControllerAdvice(new GlobalExceptionController()).build();
+
+        testResultDTOS = new ArrayList<>();
+        FROM_DATE = format.parse("26/02/2018");
```

Comment:
setting static value in non-static method is error-prone practice. Declare fields as non-static.